### PR TITLE
[dev] Add compatibility package provides to several packages

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Our package names/subpackage divisions diverge from other RPM-based distributions for various reasons. To promote compatibility with specs from other distros, we add provides statements to our RPMs to reflect alternate names where appropriate. When convenient, specs were also linted.

###### Change Log  

- `boost`: Add signals subpackage provides from devel subpackage
- `clamav`: Add provides for devel. lib subpackages from base package
- `dtc`: Add compatibility provides for libfdt, libfdt-static, libfdt-devel, python3-libfdt packages
- `efivar`: Add provides for libs subpackage from base package
- `fuse`: Add provides for libs subpackage from base package
- `gdb`: Add provides for gdbserver subpackage from base package
- `gnutls`: Add provides for c++ subpackage from base package
- `libcap-ng`: Add provides for utils subpackage from base package
- `libjpeg-turbo`: Add provides for turbojpeg, turbojpeg-devel, libjpeg-turbo-utils packages
- `libqmi`: Add provides for utils subpackage from base package
- `lm-sensors`: Add provides for libs subpackage from base package
- `net-snmp`: Add provides for libs, agent-libs subpackages from base package
- `openldap`: Add provides for clients, compat subpackages from base package
- `openscap`: Add provides for scanner subpackage from base package. Package changed to use Python 3 instead of Python 2.
- `pcre`: Add static libraries to the build and add compatibility provides for static subpackage.
- `pcre2`: Add provides for utf16, utf32 subpackages from corresponding subpackages
- `unbound`: Add python3 module support. Add provides for libs subpackage from base package

###### Does this affect the toolchain?  

**YES** (`pcre` package)

###### Test Methodology

- Local build of affected packages

### CODE REVIEW

**Review by PawelWMS — APPROVED** [review]

File: SPECS/libcap-ng/libcap-ng.spec [hunk 1]
```diff
@@ -1,70 +1,50 @@
...
-
-%description -n python2-libcap-ng
-The python2-libcap-ng package contains the python2 bindings for libcap-ng.
-
-%package  -n    python3-libcap-ng
+%package -n     python3-libcap-ng
 Summary:        Python3 bindings for libaudit
-License:        LGPLv2+
-BuildRequires:  python3-devel
-BuildRequires:  python3-libs
```
[the remark this answers is not in the document]
Comment (thcrain-msft) [reply SPECS/libcap-ng/libcap-ng.spec:33]: `python3-devel` requires `python3`, which itself requires `python3-libs`. I did this to simplify the spec and remove an edge from our build system's package graph. Every edge counts!

File: SPECS/gnutls/gnutls.spec [hunk 2]
```diff
@@ -2,45 +2,43 @@ Summary:        The GnuTLS Transport Layer Security Library
...
+Requires:       libtasn1
+Requires:       nettle >= 3.7.2
+Requires:       openssl
 Provides:       %{name}-utils = %{version}-%{release}
-
-Patch0:         CVE-2020-24659.patch
-Patch1:         CVE-2021-20231.patch
-Patch2:         CVE-2021-20232.patch
+Provides:       %{name}-c++ = %{version}-%{release}
+%if %{with_check}
```
[the remark this answers is not in the document]
Comment (thcrain-msft) [reply SPECS/gnutls/gnutls.spec:29]: Done, thanks!

File: SPECS/net-snmp/net-snmp.spec [hunk 3]
```diff
@@ -2,33 +2,34 @@
...
 BuildRequires:  systemd
-%if %{with_check}
-BuildRequires:  net-tools
-%endif
 Requires:       systemd
 Requires:       perl(:MODULE_COMPAT_%(eval "`perl -V:version`"; echo $version))
 Provides:       %{name}-utils = %{version}-%{release}
+Provides:       %{name}-libs = %{version}-%{release}
+Provides:       %{name}-agent-libs = %{version}-%{release}
+%if %{with_check}
```
[the remark this answers is not in the document]
Comment (thcrain-msft) [reply SPECS/net-snmp/net-snmp.spec:22]: Done, thanks!

File: SPECS/openscap/openscap.spec [hunk 4]
```diff
@@ -91,7 +90,7 @@ find %{buildroot} -type f -name "*.la" -delete -print
 %exclude %{_prefix}/src/debug
 %exclude %{_libdir}/debug
 %{_bindir}/*
-#%{_libexecdir}/*
+#%%{_libexecdir}/*
```
[the remark this answers is not in the document]
Comment (thcrain-msft) [reply SPECS/openscap/openscap.spec:93]: Done, thanks!

File: SPECS/openscap/openscap.spec [hunk 5]
```diff
@@ -109,46 +108,51 @@ find %{buildroot} -type f -name "*.la" -delete -print
...
+%files -n python3-%{name}
 %defattr(-,root,root)
-%{_libdir}/python2.7/*
+%{python3_sitelib}/*
 
 %changelog
+* Fri Jul 23 2021 Thomas Crain <thcrain@microsoft.com> - 1.3.1-7
+- Add provides for scanner subpackage from base package
+- Remove openscap-python subpackage
+- Add python3-openscap subpackage
```
[the remark this answers is not in the document]
Comment (thcrain-msft) [reply SPECS/openscap/openscap.spec:119]: Added more detail to the second comment to reflect this, thanks!

[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/openscap/openscap.spec:118]: Oh, I think I just missed that, but thank you for clarifying!:P

**Pull request merged**